CMakeLists.txt

```diff
@@ -533,6 +533,9 @@ target_compile_definitions(
   ${PROJECT_NAME}
   PUBLIC
   BACNET_PROTOCOL_REVISION=${BACNET_PROTOCOL_REVISION}
+  INTRINSIC_REPORTING=1
+  BINARY_INPUT_INTRINSIC_REPORTING=1
```

Comment:
I like BINARY_INPUT_INTRINSIC_REPORTING and BINARY_VALUE_INTRINSIC_REPORTING for more granular approach. However, INTRINSIC_REPORTING already exists and implies that all the objects intrinsic reporting features are enabled.

Suggestion: use INTRINSIC_REPORTING in Makefile or CMakeLists.txt, and in the bv.h and bi.h file we have another ifdef:
```
#if defined(INTRINSIC_REPORTING)
    #define BINARY_INPUT_INTRINSIC_REPORTING
#endif
```

Reply:
Excellent suggestion. How about #if defined(INTRINSIC_REPORTING) && (BINARY_VALUE_INTRINSIC_REPORTING) approach?

Reply:
Since INTRINSIC_REPORTING has the expectation of enabling NC + AI + AV event notifications, it would break builds - for developers using INTRINSIC_REPORTING solely.  

Maybe change the object specific defines to require a 1, or a 0, using 1 with not defined, and add in each .h file. Even better would be to couple it with INTRINSIC_REPORTING, and then only use the object-specific define inline in code.
```
#ifdef INTRINSIC_REPORTING
#ifndef  BINARY_INPUT_INTRINSIC_REPORTING
#define BINARY_INPUT_INTRINSIC_REPORTING 1
#endif
#endif
```

Reply:
O re-worked it so we would have a working code at which we could look. It's backwards compatible with the sole use of INTRINSIC_REPORTING flag. Next step I would remove the flags from CMakeLists.txt. I still am unsure  where would be the best place to put the BINARY_[INPUT|VALUE]_INTRINSIC_REPORTING=0  for a highlight to the users with perhaps a relevant comment.